Hepmc download URLs have changed

When trying to install version nov22p1 on a Debian 12 system using the legacy installer, I run into this error:

```
416.1 [ 30%] Creating directories for 'hepmc'
416.3 [ 34%] Performing download step (download, verify and extract) for 'hepmc'
416.9 CMake Error at /fairsoft/build/Stamp/hepmc/hepmc-download-RelWithDebInfo.cmake:37 (message):
416.9   Command failed: 1
416.9 
416.9    '/usr/bin/cmake' '-Dmake=' '-Dconfig=' '-P' '/fairsoft/build/Stamp/hepmc/hepmc-download-RelWithDebInfo-impl.cmake'
416.9 
416.9   See also
416.9 
416.9     /fairsoft/build/Log/hepmc-download.log
416.9 
416.9 
416.9 -- Log output is:
416.9 -- Downloading...
416.9    dst='/fairsoft/build/Download/hepmc/hepmc2.06.11.tgz'
416.9    timeout='none'
416.9    inactivity timeout='none'
416.9 -- Using src='https://hepmc.web.cern.ch/hepmc/releases/hepmc2.06.11.tgz'
416.9 -- [download 0% complete]
416.9 CMake Error at /fairsoft/build/Stamp/hepmc/download-hepmc.cmake:170 (message):
416.9   Each download failed!
416.9 
416.9     error: downloading 'https://hepmc.web.cern.ch/hepmc/releases/hepmc2.06.11.tgz' failed
416.9           status_code: 22
416.9           status_string: "HTTP response code said error"
416.9           log:
416.9           --- LOG BEGIN ---
416.9             Trying 188.185.5.88:443...
416.9 
416.9   Connected to hepmc.web.cern.ch (188.185.5.88) port 443 (#0)
...
416.9   The requested URL returned error: 404
416.9 
416.9   Closing connection 0
```

I did a quick research, and it turns out, hepmc download files are now located at:
[https://hepmc.web.cern.ch/releases/](https://hepmc.web.cern.ch/releases/)

Could you fix that also in the in "older" versions, like nov22 etc. ?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 ################################################################################
-# Copyright (C) 2019-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  #
+# Copyright (C) 2019-2024 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  #
 #                                                                              #
 #              This software is distributed under the terms of the             #
 #              GNU Lesser General Public Licence (LGPL) version 3,             #
@@ -33,7 +33,7 @@ set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 include(${BUILD_METHOD})
 
 find_program(BASH bash HINTS "/bin")
-find_program(SINGULARITY singularity)
-if(BASH AND SINGULARITY)
+find_program(APPTAINER apptainer)
+if(BASH AND APPTAINER)
   add_subdirectory(test/container)
 endif()

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -145,28 +145,8 @@ pipeline {
             """)
           }
 
-          if (env.CHANGE_ID != null) {
-              specs_list = [
-                [os: 'macOS'],
-              ];
-          } else {
-              specs_list = [
-                [os: 'macos-11-x86_64'],
-                [os: 'macos-12-x86_64'],
-                [os: 'macos-12-arm64']
-              ];
-          }
-
-          def macos_jobs = jobMatrix('macos', ctestcmd, specs_list)
-          { spec, label, jobsh, ctest ->
-            sh """
-              hostname -f
-              export LABEL="${label}"
-              ${ctest}
-            """
-          }
           throttle(['long']) {
-            parallel(linux_jobs + macos_jobs)
+            parallel(linux_jobs)
           }
         }
       }

--- a/cmake/legacy.cmake
+++ b/cmake/legacy.cmake
@@ -242,7 +242,7 @@ ExternalProject_Add(pythia6
 list(APPEND packages hepmc)
 set(hepmc_version "2.06.11")
 ExternalProject_Add(hepmc
-  URL https://hepmc.web.cern.ch/hepmc/releases/hepmc${hepmc_version}.tgz
+  URL https://hepmc.web.cern.ch/releases/hepmc${hepmc_version}.tgz
   URL_HASH SHA256=86b66ea0278f803cde5774de8bd187dd42c870367f1cbf6cdaec8dc7cf6afc10
   ${CMAKE_DEFAULT_ARGS} CMAKE_ARGS
     "-Dlength:STRING=CM"

--- a/cmake/legacy.cmake
+++ b/cmake/legacy.cmake
@@ -514,7 +514,7 @@ add_custom_target(source-cache
 
 include(CTest)
 
-foreach(ver IN ITEMS 18.4 18.6 18.8)
+foreach(ver IN ITEMS 18.4 18.6 18.8 19.0)
   set(TEST_VERSION v${ver}_patches)
   configure_file(test/legacy/fairroot.sh.in ${CMAKE_BINARY_DIR}/test_fairroot_${ver}.sh @ONLY)
   add_test(NAME FairRoot_${ver}

--- a/repos/fairsoft-backports/packages/hepmc/package.py
+++ b/repos/fairsoft-backports/packages/hepmc/package.py
@@ -8,8 +8,8 @@ class Hepmc(CMakePackage):
     """The HepMC package is an object oriented, C++ event record for
        High Energy Physics Monte Carlo generators and simulation."""
 
-    homepage = "https://hepmc.web.cern.ch/hepmc/"
-    url      = "https://hepmc.web.cern.ch/hepmc/releases/hepmc2.06.11.tgz"
+    homepage = "https://hepmc.web.cern.ch/"
+    url      = "https://hepmc.web.cern.ch/releases/hepmc2.06.11.tgz"
 
     tags = ['hep']
 
@@ -38,5 +38,5 @@ class Hepmc(CMakePackage):
         if version <= Version("2.06.08"):
             url = "http://lcgapp.cern.ch/project/simu/HepMC/download/HepMC-{0}.tar.gz"
         else:
-            url = "https://hepmc.web.cern.ch/hepmc/releases/hepmc{0}.tgz"
+            url = "https://hepmc.web.cern.ch/releases/hepmc{0}.tgz"
         return url.format(version)

--- a/test/container/CMakeLists.txt
+++ b/test/container/CMakeLists.txt
@@ -28,7 +28,7 @@ function(fs_spack_container)
 
   add_custom_command(OUTPUT ${target}
     COMMAND ${CMAKE_COMMAND} -E env ${BASH} -c
-    "${SINGULARITY} build -f -F ${target} ${CMAKE_CURRENT_SOURCE_DIR}/${def} > ${log} 2>&1"
+    "${APPTAINER} build -f -F ${target} ${CMAKE_CURRENT_SOURCE_DIR}/${def} > ${log} 2>&1"
     VERBATIM
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
     COMMENT "Building spack container ${target}, logging to ${CMAKE_CURRENT_BINARY_DIR}/${log}"

--- a/test/container/legacy/CMakeLists.txt
+++ b/test/container/legacy/CMakeLists.txt
@@ -1,5 +1,5 @@
 ################################################################################
-# Copyright (C) 2020-2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  #
+# Copyright (C) 2020-2024 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  #
 #                                                                              #
 #              This software is distributed under the terms of the             #
 #              GNU Lesser General Public Licence (LGPL) version 3,             #
@@ -36,7 +36,7 @@ function(fs_legacy_container)
 
   add_custom_command(OUTPUT ${target}
     COMMAND ${CMAKE_COMMAND} -E env ${BASH} -c
-      "${SINGULARITY} build -f -F ${target} ${def} > ${log} 2>&1"
+      "${APPTAINER} build -f -F ${target} ${def} > ${log} 2>&1"
     VERBATIM
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
     COMMENT "Building legacy container ${target}, logging to ${CMAKE_CURRENT_BINARY_DIR}/${log}"

--- a/test/test-start-container.sh
+++ b/test/test-start-container.sh
@@ -46,7 +46,7 @@ fi
 
 (
 	set -x
-	singularity exec -B"$bindmounts" "$image" bash -l -c "${ctestcmd}"
+	singularity exec --writable-tmpfs -B"$bindmounts" "$image" bash -l -c "${ctestcmd}"
 )
 retval=$?
 

--- a/test/test-start-container.sh
+++ b/test/test-start-container.sh
@@ -11,9 +11,9 @@ ctestcmd="$2"
 
 # Some default paths:
 : ${FS_INSTALLTREE_BASE:="/srv/alfaci/FairSoft/install-tree"}
-: ${SINGULARITY_CONTAINER_ROOT:="/cvmfs/fairsoft_dev.gsi.de/ci/for-fairsoft/latest/container"}
+: ${CONTAINER_ROOT:="/cvmfs/fairsoft_dev.gsi.de/ci/for-fairsoft/2022-11-24_1730/container"}
 
-image="${SINGULARITY_CONTAINER_ROOT}/${container}"
+image="${CONTAINER_ROOT}/${container}"
 bindmounts="/etc/environment,/cvmfs,$PWD"
 
 if [ -d "$FS_INSTALLTREE_BASE" ]

--- a/test/test-start-container.sh
+++ b/test/test-start-container.sh
@@ -46,7 +46,7 @@ fi
 
 (
 	set -x
-	singularity exec --writable-tmpfs -B"$bindmounts" "$image" bash -l -c "${ctestcmd}"
+	apptainer exec --writable-tmpfs -B"$bindmounts" "$image" bash -l -c "${ctestcmd}"
 )
 retval=$?
 

